go/vt/vttablet/tabletserver/planbuilder/dml.go

```diff
@@ -583,6 +587,9 @@ func analyzeOnDupExpressions(ins *sqlparser.Insert, pkIndex *schema.Index) (pkVa
 func extractColumnValues(rowList sqlparser.Values, colnum int) (sqltypes.PlanValue, bool) {
 	pv := sqltypes.PlanValue{Values: make([]sqltypes.PlanValue, len(rowList))}
 	for i := 0; i < len(rowList); i++ {
+		if len(rowList[i])-1 < colnum {
```

Comment:
Are you sure this is needed? I think this code is unreachable because if no value was specified, the colnum would have been computed as -1, and there is a higher level check that skips calling this function.

Reply:
Do you have the call stack? I still can't see how it happened.